docs/design/schedule_algorithm.md

```diff
@@ -0,0 +1,39 @@
+
+    finalScoreNodeA = (weight1 * priorityFunc1) + (weight2 * priorityFunc2)
+    
+After the scores of all nodes are calculated, the node with highest score is chosen as the host of the Pod. If there are more than one nodes with equal highest scores, a random one among them is chosen.
+
+Currently, Kuberntes scheduler provides some practical priority functions, including:
+
+- `LeastRequestedPriority`: The node is prioritized based on how much free resource (CPU and Memory) is available on it. The free resource is measured by the capacity minus the sum of the limits of all Pods already on the node, divided by the capacity.
+- `CalculateNodeLabelPriority`: The node can also be prioritized by checking if specified label exists.
+- `BalancedResourceAllocation`: This priority function tries to put the Pod on a node such that the CPU and Memory utilization rate is balanced after the Pod is deployed.
+- `CalculateSpreadPriority`: The Pods behind the same service are spread across the cluster. This is done by giving a higher score to a node if it has fewer Pods behind the same service.
+- `CalculateAntiAffinityPriority`: Similar as the priority function above, this function tries to spread the Pods behind same service among the nodes with particular label.
+
+The details of the above priority functions can be found in [plugin/pkg/scheduler/algorithm/priorities](../../plugin/pkg/scheduler/algorithm/priorities). By default, Kubernetes scheduler uses some of the above priority functions (in [/plugin/pkg/scheduler/algorithmprovider/defaults.go](../../plugin/pkg/scheduler/algorithmprovider/defaults.go)). Similar as predicates, you can combine the above priority functions and assign weight factors (positive number) to them as you want.
+
+There are two ways to do this. First is modifying the [/plugin/pkg/scheduler/algorithmprovider/defaults.go](../../plugin/pkg/scheduler/algorithmprovider/defaults.go) and compile a new kube-scheduler binary then start it. The second is (re)starting the kube-scheduler with a custom configuration file using `--policy-config-file` command. An example of the policy configuration file can be found at [/examples/scheduler-policy-config.json](../../examples/scheduler-policy-config.json)
```

Comment:
I would put a section heading here like
```## Customizing the scheduler```
and change the first sentence to "There are two ways to customize the scheduler (e.g. choose or add different predicate or priority functions)."